src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1577,44 +1830,28 @@ void BlockchainLMDB::reset()
   LOG_PRINT_L3("BlockchainLMDB::" << __func__);
   check_open();
 
-  mdb_txn_safe txn;
-  if (auto result = lmdb_txn_begin(m_env, NULL, 0, txn))
-    throw0(DB_ERROR(lmdb_error("Failed to create a transaction for the db: ", result).c_str()));
-
-  if (auto result = mdb_drop(txn, m_blocks, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_blocks: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_block_info, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_block_info: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_block_heights, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_block_heights: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_txs_pruned, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_txs_pruned: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_txs_prunable, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_txs_prunable: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_txs_prunable_hash, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_txs_prunable_hash: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_txs_prunable_tip, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_txs_prunable_tip: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_tx_indices, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_tx_indices: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_tx_outputs, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_tx_outputs: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_output_txs, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_output_txs: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_output_amounts, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_output_amounts: ", result).c_str()));
-  if (auto result = mdb_drop(txn, m_spent_keys, 0))
-    throw0(DB_ERROR(lmdb_error("Failed to drop m_spent_keys: ", result).c_str()));
+  mdb_write_handle txn{*m_write_data};
+
+  m_databases.drop(txn, source::blocks);
+  m_databases.drop(txn, source::block_info);
+  m_databases.drop(txn, source::block_heights);
+  m_databases.drop(txn, source::txs_pruned);
+  m_databases.drop(txn, source::txs_prunable);
+  m_databases.drop(txn, source::txs_prunable_hash);
+  m_databases.drop(txn, source::txs_prunable_tip);
+  m_databases.drop(txn, source::tx_indices);
+  m_databases.drop(txn, source::tx_outputs);
+  m_databases.drop(txn, source::output_txs);
+  m_databases.drop(txn, source::output_amounts);
+  m_databases.drop(txn, source::spent_keys);
```

Comment:
Are those actually necessary ? They cause a very large amount of possibly spurious diffs.

Reply:
Yes, there appears to be lots of re-written code unrelated to things commented on above. The re-writes appear to be mostly good. I've tended to avoid churning/rewriting the DB code because of the risk, but it could use some love. The refactoring appears to be "on the edges" so to speak, and not modifying the bulk of the processing code.

Reply:
This one isn't strictly necessary, but given that the _cursors_ have been changed from a struct to an actual array (to fix the undefined behaviour caused from the cast that happened there), I believe it makes sense to change this one as well to keep some uniformity between working with databases and cursors.

Also: The old code had a lot of repeated "open the database, do error handling"-code. To me, at least, abstracting that logic out is an improvement in its own right. If that is not an opinion you and others share with me I can of course undo this change.

Reply:
Where is that cast to array ?